more_itertools/more.py

```diff
@@ -3789,3 +3790,35 @@ def permutation_index(element, iterable):
         del pool[r]
 
     return index
+
+
+class countable:
+    """Wrap *iterable* and keep a count of how many items have been consumed.
+
+    The ``items_seen`` attribute starts at ``0`` and increments as the iterable
+    is consumed:
+
+        >>> iterable = map(str, range(10))
+        >>> it = countable(iterable)
+        >>> it.items_seen
```

Comment:
This name `items_seen` ... Seems alright now, but I'm wondering if someday I'd prefer a Counter.  If so, `items_seen` could mean a dict of observations, rather than an int.  Maybe `count_seen` or  `count_justseen`?

Reply:
A full Counter could certainly be useful in some cases, but can I ask that this wrapper never uses a Counter _by default_? Besides the question of hashability of the items, the wrapper loses quite a bit of its appeal to me if it were to use O(n) memory.